src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -110,58 +110,99 @@ public override int Execute()
                     {
-                        var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
-                        if (File.Exists(installStateFilePath))
+                        var installStateContents = InstallStateContents.FromPath(installStateFilePath);
+                        //  If install state has pinned workload set or manifest versions, then don't update workloads
+                        if (!string.IsNullOrEmpty(installStateContents.WorkloadVersion) || installStateContents.Manifests != null)
                         {
-                            var installStateContents = InstallStateContents.FromPath(installStateFilePath);
-                            _workloadSetVersion = installStateContents.WorkloadVersion;
+                            //  TODO: respect shouldUpdateWorkloads, or figure out update / install manifest difference in InstallingWorkloadCommand
+                            shouldUpdateWorkloads = false;
                         }
                     }
+                }
+
+                try
+                {
+                    Reporter.WriteLine();
 
                     DirectoryPath? offlineCache = string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption);
                     var workloadIds = _workloadIds.Select(id => new WorkloadId(id));
-                    if (string.IsNullOrWhiteSpace(_workloadSetVersion) && string.IsNullOrWhiteSpace(_workloadSetVersionFromGlobalJson))
+
+                    // Add workload Ids that already exist to our collection to later trigger an update in those installed workloads
+                    var installedWorkloads = _workloadInstaller.GetWorkloadInstallationRecordRepository().GetInstalledWorkloads(_sdkFeatureBand);
+                    var previouslyInstalledWorkloads = installedWorkloads.Intersect(workloadIds);
+                    if (previouslyInstalledWorkloads.Any())
                     {
-                        InstallWorkloads(
-                            workloadIds,
-                            _skipManifestUpdate,
-                            _includePreviews,
-                            offlineCache);
+                        Reporter.WriteLine(string.Format(LocalizableStrings.WorkloadAlreadyInstalled, string.Join(" ", previouslyInstalledWorkloads)).Yellow());
                     }
-                    else
+                    workloadIds = workloadIds.Concat(installedWorkloads).Distinct();
+                    workloadIds = WriteSDKInstallRecordsForVSWorkloads(workloadIds);
+
+                    if (!_skipManifestUpdate)
                     {
-                        RunInNewTransaction(context =>
+                        var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
+                        if (string.IsNullOrWhiteSpace(_fromRollbackDefinition) &&
```

Comment:
Shouldn't there be more conditions here for other reasons to require a manifest udpate?

Reply:
clarification: consider adding string.isnull checks for --version inputs and global.json version inputs to be complete

Reply:
Ping on this one

Reply:
Are you saying that you think we should skip manifest update if `--sdk-version` is specified or if there's a workload version in global.json?  In those cases we do want to call the `UpdateWorkloadManifests` method.  That method is responsible for respecting the version from `--sdk-version` or global.json.  If a version was specified in either of those ways, then it will install those versions.

So I think the logic is correct, let me know if there's a case I'm missing.

Reply:
> we should skip manifest update if `--sdk-version` is specified or if there's a workload version in global.json

The opposite: I think we should never skip manifest updates if either of those is specified. This if statement currently says that if rollback is not specified, and there is something pinned in the install state file, then we do skip manifest updates, which is wrong.

Reply:
```suggestion
                        if (string.IsNullOrWhiteSpace(_fromRollbackDefinition) &&
                            !SpecifiedWorkloadSetVersionOnCommandLine &&
                            !SpecifiedWorkloadSetVersionInGlobalJson &&
```

Reply:
The explanation as I understand it is that Nathan is saying that if we already have an install state, this logic will skip workload updates even if you have a global.json or --versino specified (which could be different than what is in the install state). So I think you are agreeing with each other. Daniel, I think you're implying that UpdateWorkloadManifests would end up getting called for those cases which I think is true the first time a customer does it but any subsequent times after having an install state, I think forgind may be right unless I'm missing a code path that calls it elsewhere.

Reply:
Thanks, I think I get it now.  I will try to add a test to validate the behavior and then update the logic.  I'm not sure how quickly I'll be able to do that though.

Reply:
Should be fixed now